### PR TITLE
Checking if scene has been loaded before we try to search it for extra playspaces

### PR DESCRIPTION
**Pull request by marek-stoj**
## Overview

Fixes an issue with an exception being thrown from within MRTK when opening the project:
```
ArgumentException: The scene is not loaded.
UnityEngine.SceneManagement.Scene.GetRootGameObjects (System.Collections.Generic.List`1[T] rootGameObjects) (at C:/buildslave/unity/build/Runtime/Export/SceneManager/Scene.cs:91)
UnityEngine.SceneManagement.Scene.GetRootGameObjects () (at C:/buildslave/unity/build/Runtime/Export/SceneManager/Scene.cs:75)
Microsoft.MixedReality.Toolkit.MixedRealityPlayspace.EditorSceneManagerSceneOpened (UnityEngine.SceneManagement.Scene scene, UnityEditor.SceneManagement.OpenSceneMode mode) (at Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs:205)
UnityEditor.SceneManagement.EditorSceneManager.Internal_SceneOpened (UnityEngine.SceneManagement.Scene scene, UnityEditor.SceneManagement.OpenSceneMode mode) (at C:/buildslave/unity/build/Editor/Mono/SceneManagement/EditorSceneManager.cs:47)
UnityEditor.SceneManagement.EditorSceneManager:OpenScene(String, OpenSceneMode)
Microsoft.MixedReality.Toolkit.Utilities.EditorSceneUtils:LoadScene(SceneInfo, Boolean, Scene&) (at Assets/MixedRealityToolkit/Utilities/Scenes/EditorSceneUtils.cs:188)
Microsoft.MixedReality.Toolkit.SceneSystem.MixedRealitySceneSystem:EditorUpdateManagerScene() (at Assets/MixedRealityToolkit.Services/SceneSystem/MixedRealitySceneSystemEditorOperations.cs:471)
Microsoft.MixedReality.Toolkit.SceneSystem.<EditorCheckForChanges>d__166:MoveNext() (at Assets/MixedRealityToolkit.Services/SceneSystem/MixedRealitySceneSystemEditorOperations.cs:307)
System.Runtime.CompilerServices.AsyncVoidMethodBuilder:Start(<EditorCheckForChanges>d__166&)
Microsoft.MixedReality.Toolkit.SceneSystem.MixedRealitySceneSystem:EditorCheckForChanges()
Microsoft.MixedReality.Toolkit.SceneSystem.MixedRealitySceneSystem:EditorOnInitialize() (at Assets/MixedRealityToolkit.Services/SceneSystem/MixedRealitySceneSystemEditorOperations.cs:151)
Microsoft.MixedReality.Toolkit.SceneSystem.MixedRealitySceneSystem:Initialize() (at Assets/MixedRealityToolkit.Services/SceneSystem/MixedRealitySceneSystem.cs:162)
Microsoft.MixedReality.Toolkit.<>c:<InitializeAllServices>b__60_0(IMixedRealityService) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:923)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit:ExecuteOnAllServicesInOrder(Action`1) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:1019)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit:InitializeAllServices() (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:923)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit:InitializeServiceLocator() (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:468)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit:InitializeInstance() (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:582)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit:RegisterInstance(MixedRealityToolkit, Boolean) (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:720)
Microsoft.MixedReality.Toolkit.MixedRealityToolkit:OnValidate() (at Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs:1456)
```

### CODE REVIEW

File: Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs [hunk 1]
```diff
@@ -200,7 +200,10 @@ private static void EditorSceneManagerSceneOpened(Scene scene, OpenSceneMode mod
             }
             else
             {
-                SearchForAndDisableExtraPlayspaces(scene.GetRootGameObjects());
+                if (scene.isLoaded)
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs]: Do you think this should even be a little higher, alongside the `if (Application.isPlaying)` check? I wonder if we should be doing anything at all in this method if the scene isn't actually loaded. @Railboy?

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/81383453589488e2df92d54c3938cab5e57bc424/Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs#L192

Comment (marek-stoj) [reply Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs]: Good point. I was just going for a minimum-impact change.

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs]: I believe @keveleigh is correct here. Let's move the check to line 192.

Nice fix!

**Comment by davidkline-ms** [comment]
@marek-stoj, can you resolve the merge conflict with MixedRealityPlayspace.cs? 

Thanks!

File: Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs (hunk 1 above)
Comment (marek-stoj) [reply Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs]: Actually, EditorSceneUtils.GetRootGameObjectsInLoadedScenes() iterates over all open scenes but skips those which are not yet loaded. So I'm thinking that maybe the check should stay where it is now (?) In other words - the fact that the scene the event pertains to isn't loaded, won't affect the path where mixedRealityPlayspace is null.

**Comment by marek-stoj** [comment]
Hmm, I did an ff-only merge of upstream/mrtk_development into my fork and then a rebase of the fix but now the PR wants to merge all those commits. That can't be right, can it?

**Comment by keveleigh** [comment]
@marek-stoj Whoops! I'd try (assuming you have the latest mrtk_development fetched locally):

`git reset --hard mrtk_development`
`git cherry-pick 40d8563b8f555e5b9268ce47201b64370c4564ab`
`git push -f`

Would be worth double checking the state of things before force pushing with that last command, but that should get you back into working order!

**Comment by davidkline-ms** [comment]
looks like you picked up the tree reworking change and git is a bit confused.

**Comment by marek-stoj** [comment]
Got it back in order. Thanks @keveleigh!

**Pull request merged**